fs/_ftp_parse.py

```diff
@@ -24,9 +24,9 @@
     \s+?
     (\d+)
     \s+?
-    ([\w\-]+)
+    ([A-Za-z0-9][A-Za-z0-9\-\.\_\@]*[\$]{0,1})
```

Comment:
You can replace `[\$]{0,1}` with just `\$?` here.

Reply:
Yep!